src/helpers/getByAPI.js

```diff
@@ -36,17 +36,37 @@ const getNodeByText = (node, text) => {
 };
 
 const getChildrenAsText = children => {
-  return React.Children.map(children, child => {
+  let textContent = [];
+
+  React.Children.map(children, child => {
     if (typeof child === 'string') {
-      return child;
+      return textContent.push(child);
     }
 
     if (typeof child === 'number') {
-      return child.toString();
+      return textContent.push(child.toString());
     }
 
-    return '';
+    if (child.props.children) {
```

Comment:
Here, `child` can be `NaN`, `null`, `undefined`, `Symbol`. None of these have `props` member. We should guard against it

Reply:
This will also check only one level of text, and you can nest more or components that wrap Text host component. If we want to get it right, I feel we should make `getChildrenAsText` recursive